cmd/mungectl/cmd/root.go

```diff
@@ -32,10 +34,26 @@ var rootCmd = &cobra.Command{
 }
 
 func init() {
+	cobra.OnInitialize(initConfig)
 	rootCmd.CompletionOptions.DisableDefaultCmd = true
 	rootCmd.AddCommand(key.KeyCmd)
 }
 
+// Initialize mungectl configuration.
+func initConfig() {
+	viper.SetConfigName("mungectl")
+	viper.SetConfigType("yaml")
+
+	if os.Getenv("SNAP") != "" {
+		common := os.Getenv("SNAP_COMMON")
+		viper.AddConfigPath(path.Join(common, "/etc/munge/mungectl.yaml"))
+		viper.SetDefault("keyfile", path.Join(common, "/etc/munge/munge.key"))
+	} else {
+		viper.AddConfigPath("/etc/munge/mungectl.yaml")
+		viper.SetDefault("keyfile", "/etc/munge/munge.key")
+	}
```

Comment:
Are we planning to add more settings on `mungectl.yaml`? Because if not, then having a whole file just to set a path it's a bit overkill.

Also, do we need to handle snap in a special way? I think having a `MUNGECTL_KEY/CONFIG` env var should cover the snap use case, defaulting to `/etc/munge/mungectl.yaml/key` if unset.

Reply:
It depends on how far we want to take `mungectl`. I could see us adding more configuration options to `mungectl.yaml` if we need different startup parameters for munged if we wanted. The _mungectl.yaml_ file is not required to exist, it's just an available option if _someone_ decides they want to use it. The default will be `/etc/munge/munge.key` or `$SNAP_COMMON/etc/munge/munge.key` if the file does not exist.

For example, gambol has couple different possible configuration locations:

https://github.com/NucciTheBoss/gambol/blob/431d81f1f3c4bf34117a4579ba1e69d7f8f04757/cmd/gambol/cmd/root.go#L57-L59

I was debating using the environment variable, so I can do that instead. I mostly just went with the current option since it's pretty clear that we're going to use this tool within snap so snap support should be a first class citizen :sweat_smile: